cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/PrometheusMetricSamplerTest.java

```diff
@@ -0,0 +1,491 @@
+        _prometheusMetricSampler.configure(config);
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testConfigureWithNoPrometheusEndpointFails() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        addCapacityConfig(config);
+        _prometheusMetricSampler.configure(config);
+    }
+
+    @Test(expected = SamplingException.class)
+    public void testGetSamplesQueryThrowsException() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "http://kafka-cluster-1.org:9090");
+        addCapacityConfig(config);
+        _prometheusMetricSampler.configure(config);
+
+        MetricSamplerOptions metricSamplerOptions = buildMetricSamplerOptions();
+        _prometheusMetricSampler._prometheusAdapter = _prometheusAdapter;
+        expect(_prometheusAdapter.queryMetric(anyString(), anyLong(), anyLong()))
+            .andThrow(new IOException("Exception in fetching metrics"));
+
+        replay(_prometheusAdapter);
+        _prometheusMetricSampler.getSamples(metricSamplerOptions);
+    }
+
+    @Test
+    public void testGetSamplesCustomPrometheusQuerySupplier() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "http://kafka-cluster-1.org:9090");
+        addCapacityConfig(config);
+        config.put(PROMETHEUS_QUERY_SUPPLIER_CONFIG, TestQuerySupplier.class.getName());
+        _prometheusMetricSampler.configure(config);
+
+        MetricSamplerOptions metricSamplerOptions = buildMetricSamplerOptions();
+        _prometheusMetricSampler._prometheusAdapter = _prometheusAdapter;
+
+        expect(_prometheusAdapter.queryMetric(eq(TestQuerySupplier.TEST_QUERY), anyLong(), anyLong()))
+            .andReturn(buildBrokerResults());
+        replay(_prometheusAdapter);
+
+        _prometheusMetricSampler.getSamples(metricSamplerOptions);
+
+        verify(_prometheusAdapter);
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testGetSamplesPrometheusQuerySupplierUnknownClass() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "http://kafka-cluster-1.org:9090");
+        addCapacityConfig(config);
+        config.put(PROMETHEUS_QUERY_SUPPLIER_CONFIG, "com.test.NonExistentClass");
+        _prometheusMetricSampler.configure(config);
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testGetSamplesPrometheusQuerySupplierInvalidClass() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put(PROMETHEUS_SERVER_ENDPOINT_CONFIG, "http://kafka-cluster-1.org:9090");
+        addCapacityConfig(config);
+        config.put(PROMETHEUS_QUERY_SUPPLIER_CONFIG, String.class.getName());
+        _prometheusMetricSampler.configure(config);
+    }
+
+    private MetricSamplerOptions buildMetricSamplerOptions() {
```

Comment:
Nit: Can we make this builder (and in general other builders in this class) static?

Reply:
Done.